File: employee-management-frontend/src/types/task.ts
export type TaskStatus = 'PENDING' | 'IN_PROGRESS' | 'COMPLETED'
export type TaskPriority = 'HIGH' | 'MEDIUM' | 'LOW'

export interface Task {
  id: number
  title: string
  description: string
  status: TaskStatus
  progress: number
  priority?: TaskPriority
  remarks: string
  deadline: string
  assignedDate?: string
  dueDate?: string
  completedDate?: string | null
  estimatedHours?: number | null
  employeeId?: number | null
  employeeName?: string | null
  projectId: number
  projectName?: string | null
}

export type TaskInput = Omit<Task, 'id' | 'employeeName' | 'projectName'>

export interface TaskSearchParams {
  status?: string
  deadline?: string
}

export interface AssignTaskRequest {
  employeeId: number
  projectId: number
}
export interface UpdateProgressRequest { progress: number }
export interface UpdateStatusRequest { status: TaskStatus }
export interface UpdateRemarksRequest { remarks: string }

File: employee-management-backend/src/main/java/com/example/employeemanagement/entity/Task.java
package com.example.employeemanagement.entity;

import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDate;

@Entity
@Table(name = "tasks")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Task {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String title;

    @Column(length = 1000)
    private String description;

    private String status;

    private Integer progress;

    private String priority;

    private String remarks;

    private LocalDate deadline; // Keep deadline as alias or map to dueDate

    private LocalDate assignedDate;

    private LocalDate dueDate;

    private LocalDate completedDate;

    private Integer estimatedHours;

    @ManyToOne
    @JoinColumn(name = "employee_id")
    private Employee employee;

    @ManyToOne
    @JoinColumn(name = "project_id")
    private Project project;
}